src/details/ArborX_DetailsDistributedTreeNearestHelpers.hpp

```diff
@@ -93,6 +93,76 @@ struct WithinDistanceFromPredicates
   Predicates predicates;
   Distances distances;
 };
+
+namespace Dispatch
+{
+
+template <typename Tag, typename Geometry>
+struct approx_expand_by_radius;
+
+template <typename Point>
+struct approx_expand_by_radius<PointTag, Point>
+{
+  template <typename Float>
+  KOKKOS_FUNCTION static auto apply(Point const &point, Float r)
+  {
+    constexpr int DIM = GeometryTraits::dimension_v<Box>;
+    using Coordinate = GeometryTraits::coordinate_type_t<Point>;
+    auto const &hyper_point = reinterpret_cast<
+        ExperimentalHyperGeometry::Point<DIM, Coordinate> const &>(point);
+    return ExperimentalHyperGeometry::Sphere<DIM, Coordinate>{hyper_point, r};
+  }
+};
+
+template <typename Box>
+struct approx_expand_by_radius<BoxTag, Box>
+{
+  template <typename Float>
+  KOKKOS_FUNCTION static auto apply(Box const &box, Float r)
+  {
+    Box new_box = box;
+    auto &min_corner = new_box.minCorner();
+    auto &max_corner = new_box.maxCorner();
+    constexpr int DIM = GeometryTraits::dimension_v<Box>;
+    for (int d = 0; d < DIM; ++d)
+    {
+      min_corner[d] -= r;
+      max_corner[d] += r;
+    }
+    return new_box;
+  }
+};
+
+template <typename Sphere>
+struct approx_expand_by_radius<SphereTag, Sphere>
+{
+  template <typename Float>
+  KOKKOS_FUNCTION static auto apply(Sphere const &sphere, Float r)
+  {
+    return Sphere{sphere.centroid(), sphere.radius() + r};
+  }
+};
+
+template <typename Ray>
+struct approx_expand_by_radius<RayTag, Ray>
+{
+  template <typename Float>
+  KOKKOS_FUNCTION static auto const &apply(Ray const &ray, Float)
+  {
+    return ray;
+  }
+};
```

Comment:
Shouldn't that modify its origin?

Reply:
No. Essentially, we want to find the ranks of the top tree bounding boxes that are closer than a specified distance (found in phase I). As the distance for the ray actually means the distance from ray's origin along the ray itself, and not in every dimension, it does not require changing the origin.